src/component/pokemon/Pokemon.js

```diff
@@ -0,0 +1,77 @@
+import React, {useEffect, useState} from "react";
+import axios from "axios";
+import "./Pokemon.css";
+import {Link, useNavigate} from "react-router-dom";
+
+
+const Pokemon = ({name}) => {
+    const [ loading, setLoading ] = useState( false );
+    const [ error, setError ] = useState( false );
+    const [pokemon, setPokemon] = useState([]);
+
+
+
+
+
+    useEffect( () => {
+        const controller = new AbortController();
+
+        const fetchData = async () => {
+
+
+                setLoading( true );
+            try {
+                setError( false );
+                const response = await axios.get( `https://pokeapi.co/api/v2/pokemon/${name}`, {
+                    signal: controller.signal,
+                } );
+                setPokemon( response.data );
+
+
+            } catch ( e ) {
+                setError( true )
+
+                if(axios.isCancel(e)){
+                    console.log('The axios request was cancelled')
+                } else {
+                    console.error(e)
+                }
+            }
+            setLoading( false );
+        }
+        fetchData();
+
+
+        return function cleanup() {
+            controller.abort();
+        }
+    }, [name] )
+
+
+    return (
+        <>
+            { loading && <p>Loading...</p> }
+            { error && <p>Error: Could not fetch data!</p> }
+
+            {Object.keys(pokemon).length > 0 &&
```

Comment:
Dan kan deze check wat simpeler
```suggestion
            {pokemon &&
```